store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/MiniAccumuloStore.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.accumulostore;
+
+import com.google.common.io.Files;
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.core.security.SystemPermission;
+import org.apache.accumulo.minicluster.MiniAccumuloCluster;
+import org.apache.accumulo.minicluster.MiniAccumuloConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Set;
+
+/**
+ * A {@code MiniAccumuloStore} is an {@link AccumuloStore} which sets up an {@link MiniAccumuloCluster}
+ * for each unique instance name. If you create two Accumulo stores with the same instance name,
+ * the same cluster will be used for both. It's advisable to re-use an instance so that you don't
+ * spend unnecessary time spinning up mini-clusters.
+ *
+ * It's dependencies mean it cannot be run in a REST API.
+ *
+ * If a user hasn't been created on the accumulo instance, it will be created for you with the
+ * CREATE_NAMESPACE and CREATE_TABLE permissions.
+ *
+ * If you specify the authorisations in the store properties, the current authorisations for any
+ * existing user will be overwritten.
+ */
+public class MiniAccumuloStore extends AccumuloStore {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MiniAccumuloStore.class);
+
+    private static final HashMap<String, MiniAccumuloCluster> CLUSTER_INSTANCES = new HashMap<>();
+    private static final HashMap<String, File> ACCUMULO_DIRECTORIES = new HashMap<>();
```

Comment:
Typically mutable collections don't use constant casing

Reply:
Yeah I agree it isn't really a constant. Only reason it was like this is because checkstyle enforces it. I can remove the rule?

Reply:
Had a dig around and there's no easy fix for this. There's been an issue for most nuance to this rule on the Checkstyle project since 2013, so I don't think it's going to be resolved any time soon. Given the scope of the variables, it's probably better to keep with the checkstyle as generally it's a useful check.